crates/dojo/bindgen/src/plugins/unrealengine/mod.rs

```diff
@@ -0,0 +1,1376 @@
+                            .inners
+                            .iter()
+                            .map(|field| {
+                                format!(
+                                    "strings.Append(ConvertToFeltHexa<{cpp_type}>(value.{name}, \
+                                     \"{type_name}\"));",
+                                    cpp_type = Self::map_type(&field.token),
+                                    name = Self::to_pascal_case(&field.name),
+                                    type_name = &field.token.type_name(),
+                                )
+                            })
+                            .collect::<Vec<String>>()
+                            .join("\n            ")
+                    );
+                }
+            }
+
+            // Handle enum dependencies
+            let enum_keys: Vec<String> = handled_tokens
+                .iter()
+                .filter(|(_, s)| {
+                    model_struct.inners.iter().any(|inner| {
+                        s.r#type == CompositeType::Enum
+                            && check_token_in_recursively(&inner.token, &s.type_name())
+                    })
+                })
+                .map(|(k, _)| k.clone())
+                .collect();
+
+            for key in enum_keys {
+                if let Some(s) = handled_tokens.remove(&key) {
+                    if !defined_enums.contains(&s.type_name()) {
+                        static_converters += &format!(
+                            "static ED{type_name} ConvertToED{type_name}(const Member* member) {{
+        return static_cast<ED{type_name}>(member->ty->primitive.u8);
+    }}
+
+    ",
+                            type_name = s.type_name()
+                        );
+                        else_if_type_converter += &format!(
+                            "else if constexpr (std::is_same_v<T, ED{type_name}>) {{
+        if (strcmp(expectedType, \"{type_name}\") == 0) {{
+            output = TypeConverter::ConvertToED{type_name}(member);
+        }}
+    }}
+    ",
+                            type_name = s.type_name()
+                        );
+
+                        convert_custom_types_to_felt_string += &format!(
+                            "else if constexpr (std::is_same_v<T, {cpp_type}>) {{
+        if (strcmp(valueType, \"{type_name}\") == 0) {{
+            FString hexValue = FString::Printf(TEXT(\"%X\"), static_cast<int>(value));
+            while (hexValue.Len() < 64) {{
+                hexValue = TEXT(\"0\") + hexValue;
+            }}
+            return TArray<FString>{{TEXT(\"0x\") + hexValue}};
+        }}
+    }}
+    ",
+                            cpp_type = "ED".to_owned() + &s.type_name(),
+                            type_name = s.type_name()
+                        );
+                        defined_enums.insert(s.type_name());
+                    }
+                }
+            }
+        }
+
+        format!(
```

Comment:
Combining raw string (`r#`) and `format!` could avoid escaping the double quotes.